internal/controller/operatorconfigmap_controller.go

```diff
@@ -411,13 +428,20 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI() error {
 			return fmt.Errorf("failed to own csi operator config: %v", err)
 		}
 		templates.CSIOperatorConfigSpec.DeepCopyInto(&csiOperatorConfig.Spec)
-		csiOperatorConfig.Spec.DriverSpecDefaults.ImageSet = &corev1.LocalObjectReference{Name: cmName}
-		csiOperatorConfig.Spec.DriverSpecDefaults.ClusterName = ptr.To(string(clusterVersion.Spec.ClusterID))
+		driverSpecDefaults := csiOperatorConfig.Spec.DriverSpecDefaults
```

Comment:
This will create a copy and will not mutate the original struct.
```suggestion
		driverSpecDefaults := &csiOperatorConfig.Spec.DriverSpecDefaults
```